extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php

```diff
@@ -20,27 +16,75 @@ public function __construct( $cityId ) {
 		$this->cityId = $cityId;
 	}
 
-	// TODO: Refactor this to use Swagger, see JPN-631.
-	private function getRequestUrl( $showLatest, $limit ) {
+	// TODO: Consider changing this request to use Swagger when unblocked. See JPN-631
+	private function apiRequest( $url ) {
+		$data = Http::get( $url );
+		$obj = json_decode( $data, true );
+		return $obj;
+	}
+
+	private function getCategoryRequestUrl() {
+		global $wgDevelEnvironment;
+		if ( empty( $wgDevelEnvironment ) ) {
+			return self::DISCUSSIONS_API_BASE . "$this->cityId/forums?responseGroup=small&viewableOnly=true";
+		}
+		return self::DISCUSSIONS_API_BASE_DEV . "$this->cityId/forums?responseGroup=small&viewableOnly=true";
+	}
+
+	public function getCategoryId( $category ) {
+		$memcKey = wfMemcKey( __METHOD__, self::MCACHE_VER );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)